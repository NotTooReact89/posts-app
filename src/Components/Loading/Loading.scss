$spinner: #32122f;
$size: 60px;
$background: white;

.loading {
  min-height: 82vh;
  min-width: 100vw;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;

  &--inContainer {
    display: block;
    min-height: 100%;
    min-width: 100%;
  }

  &_spinner {
    width: 40px;
    height: 40px;
    position: absolute;

    &--inContainer {
      height: 100%;
      width: 100%;
      left: 0;
      top: 0;
      margin: 0;
    }
  }

  &_doubleBounce1,
  &_doubleBounce2 {
    width: 100%;
    height: 100%;
    border-radius: 50%;
    background-color: #32122f;
    opacity: 0.6;
    position: absolute;
    top: 0;
    left: 0;

    -webkit-animation: sk-bounce 2s infinite ease-in-out;
    animation: sk-bounce 2s infinite ease-in-out;

    &--white {
      background-color: #ffffff;
    }
  }

  &_doubleBounce2 {
    -webkit-animation-delay: -1s;
    animation-delay: -1s;
  }

  @-webkit-keyframes sk-bounce {
    0%,
    100% {
      -webkit-transform: scale(0);
    }
    50% {
      -webkit-transform: scale(1);
    }
  }

  @keyframes sk-bounce {
    0%,
    100% {
      transform: scale(0);
      -webkit-transform: scale(0);
    }
    50% {
      transform: scale(1);
      -webkit-transform: scale(1);
    }
  }
}
