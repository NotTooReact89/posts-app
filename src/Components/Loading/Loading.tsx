// eslint-disable @typescript/camelcase
import React from "react";
import classNames from "classnames";

import "./Loading.scss";

const Loading = ({
  isInContainer,
  customstyle,
}: {
  isInContainer?: boolean;
  customstyle?: Object;
}) => {
  const loadingClass = classNames({
    loading: true,
    "loading--inContainer": isInContainer,
  });

  const spinnerClass = classNames({
    loading_spinner: true,
    "loading_spinner--inContainer": isInContainer,
  });

  const bounce1 = classNames({
    loading_doubleBounce1: true,
    "loading_doubleBounce1--white": isInContainer,
  });

  const bounce2 = classNames({
    loading_doubleBounce2: true,
    "loading_doubleBounce2--white": isInContainer,
  });

  return (
    <div className={loadingClass} style={customstyle}>
      <div className={spinnerClass}>
        <div className={bounce1} />
        <div className={bounce2} />
      </div>
    </div>
  );
};

export default Loading;
