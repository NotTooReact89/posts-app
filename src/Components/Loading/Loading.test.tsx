import * as React from "react";
import TestRenderer from "react-test-renderer";
import Loading from "./Loading";

describe("Loading", () => {
  it("should render the component", () => {
    const component = TestRenderer.create(<Loading isInContainer />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
