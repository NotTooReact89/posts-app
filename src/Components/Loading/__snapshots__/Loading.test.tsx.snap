// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render the component 1`] = `
<div
  className="loading loading--inContainer"
>
  <div
    className="loading_spinner loading_spinner--inContainer"
  >
    <div
      className="loading_doubleBounce1 loading_doubleBounce1--white"
    />
    <div
      className="loading_doubleBounce2 loading_doubleBounce2--white"
    />
  </div>
</div>
`;
